<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="initial_ts_state_from_agent" default="False"/>
  <arg name="num_mobile" default=1/>
  <arg name="num_quadruped" default=1/>

<!--   <arg name="agent_name_mobile_1" default="nexus"/> -->
<!--   <arg name="agent_name_mobile_2" default="neuxs"/> -->
<!--   <arg name="agent_name_dog_1" default="a1"/> -->

<!--   <param name="agent_name" type="string" value="$(arg agent_name)" /> -->
  <!-- hard and soft task parameters -->
  <rosparam command="load" file="$(find ltl_automaton_planner)/config/example_ltl_formula.yaml" />

  <!-- Transition system for mobile robot-->
  <param name="transition_system_mobile_textfile" type="str" textfile="$(find ltl_automaton_nexus)/config/nexus_ts.yaml" />

  <!-- Transition system for quadruped-->
  <param name="transition_system_quadruped_textfile" type="str" textfile="$(find ltl_automation_a1)/config/a1_ts.yaml" />

  <!-- Transition system for quadruped-->
  <param name="transition_system_mobile_textfile" type="str" textfile="$(find ltl_automation_a1)/config/a1_ts.yaml" />

  <!-- Run planner node -->
  <node name="ltl_planner" pkg="ltl_automaton_planner" type="planner_node.py" output="screen">
    <param name="initial_ts_state_from_agent" type="bool" value="$(arg initial_ts_state_from_agent)" />
  </node>

  <!-- Run the gazebo world-->

  <!-- Run the mocap simulator-->
  <include file="$(find mocap_simulator)/launch/qualisys_simulator.launch" />

  <!-- Run quadruped navigation -->
  <include file="$(find ltl_automation_a1)/launch/a1_monitor_bt.launch">
    <arg name="agent_name" value="$(arg agent_name)"/>
  </include>


</launch>
